libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/roadname/view/MapboxRoadNameView.kt

```diff
@@ -0,0 +1,141 @@
+
+    /**
+     * Default view to render a road name label.
+     */
+    constructor(context: Context) : super(context)
+
+    /**
+     * Default view to render a road name label.
+     */
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
+
+    /**
+     * Default view to render a road name label.
+     */
+    constructor(
+        context: Context,
+        attrs: AttributeSet?,
+        defStyleAttr: Int
+    ) : super(context, attrs, defStyleAttr)
+
+    /**
+     * Invoke the method to render the view with road name.
+     *
+     * @param road Road
+     */
+    fun renderRoadName(road: Road) {
+        if (road.components.isNotEmpty()) {
+            roadComponents.clear()
+            roadComponents.addAll(road.components)
+        }
+        renderRoadNameLabel()
+    }
+
+    /**
+     * Invoke the method to render the view with route shields.
+     *
+     * @param expectedShields list of shields to render
+     */
+    fun renderRoadNameWith(expectedShields: List<Expected<RouteShieldError, RouteShieldResult>>) {
+        expectedShields.forEach { expected ->
+            if (expected.isValue && expected.value != null) {
+                this.shields.add(expected.value!!.shield)
+            }
+        }
+        renderRoadNameLabel()
+    }
+
+    private fun renderRoadNameLabel() {
+        val roadNameLabel = SpannableStringBuilder()
+        roadComponents.forEach { component ->
+            val routeShield = when {
+                component.shield != null || component.imageBaseUrl != null -> {
+                    val routeShield = shields.find {
+                        when (it) {
+                            is RouteShield.MapboxDesignedShield -> {
+                                it.compareWith(other = component.shield)
+                            }
+                            is RouteShield.MapboxLegacyShield -> {
+                                it.compareWith(other = component.imageBaseUrl)
+                            }
+                            else -> {
+                                false
+                            }
+                        }
+                    }
+                    routeShield
+                }
+                else -> {
+                    null
+                }
+            }
+            renderRoadName(
+                text = component.text,
+                shield = routeShield,
+                builder = roadNameLabel
+            )
+        }
+        text = roadNameLabel
+    }
+
+    private fun renderRoadName(
+        text: String,
+        shield: RouteShield?,
+        builder: SpannableStringBuilder
+    ) {
+        ifNonNull(shield) {
+            val shieldBuilder = replaceTextWithShield(
+                toReplace = text,
+                replaceWith = it
+            )
+            builder.append(shieldBuilder)
+        } ?: builder.append(text)
+        builder.append(" ")
+    }
```

Comment:
```suggestion
            roadNameLabelBuilder.renderRoadName(
                text = component.text,
                shield = routeShield,
            )
        }
        text = roadNameLabelBuilder
    }

    private fun SpannableStringBuilder.renderRoadName(
        text: String,
        shield: RouteShield?
    ) {
        ifNonNull(shield) {
            val shieldBuilder = replaceTextWithShield(
                toReplace = text,
                replaceWith = it
            )
            append(shieldBuilder)
        } ?: append(text)
        append(" ")
    }
```


Not a fan of passing instances into functions to be mutated in place, it makes it much harder to follow when reading. Suggestion to use an extension instead, or us an `apply` block with a function that applies the params to a builder.